app/src/main/java/com/tma/sparking/services/handler/ParkingFieldManager.java

```diff
@@ -0,0 +1,56 @@
+package com.tma.sparking.services.handler;
+
+import android.content.Context;
+import android.os.Handler;
+import android.os.Message;
+
+import com.tma.sparking.models.ParkingField;
+import com.tma.sparking.services.ParkingFieldService;
+import com.tma.sparking.services.http.Parking;
+
+import java.util.List;
+
+public class ParkingFieldManager {
+    private Handler mMainThreadHandler;
+    private GetParkingFieldTask mGetParkingFieldTask;
+    private GetParkingFieldCallback mGetParkingFieldCallback;
+    private long mDelayMillis;
+    private long mChannelId;
+
+    public ParkingFieldManager(Context context) {
+        mMainThreadHandler = new Handler(context.getMainLooper()) {
+            @Override
+            public void handleMessage(Message msg) {
+                if (!mGetParkingFieldTask.hasError()) {
+                    List<ParkingField> parkingFields = mGetParkingFieldTask.getParkingFields();
+                    mGetParkingFieldCallback.onParkingFieldsLoaded(parkingFields);
```

Comment:
mGetParkingFieldTask is used by 2 threads - Main Thread here and ParkingFieldHandlerThread. This needs to be synchronized.